2.4/examples/replica/mongodb-clustered-petset.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: mongodb-petset-replica
+  annotations:
+    description: "MongoDB Replication Example (based on PetSet)"
+    iconClass: "icon-mongodb"
+    tags: "database,mongodb,replication"
+parameters:
+  - name: MONGODB_USER
+    description: "The username for MongoDB account to be created"
+    generate: expression
+    from: "[a-zA-Z0-9]{3}"
+  - name: MONGODB_PASSWORD
+    description: "Password for the MongoDB user account"
+    generate: expression
+    from: "[a-zA-Z0-9]{12}"
+  - name: MONGODB_DATABASE
+    description: "Name for the MongoDB database"
+    value: sampledb
+  - name: MONGODB_ADMIN_PASSWORD
+    description: "Password for the MongoDB admin user"
+    generate: expression
+    from: "[a-zA-Z0-9]{12}"
+  - name: MONGODB_REPLICA_NAME
+    description: "The name of the replica set"
+    value: rs0
+  # It must match with Service.metadata.name and PetSet.spec.serviceName
+  - name: MONGODB_SERVICE_NAME
+    description: "The name of the MongoDB Service (used to DNS lookup)"
+    value: mongodb
+  - name: MONGODB_KEYFILE_VALUE
+    description: "The value of the MongoDB Key (see for details: https://docs.mongodb.org/v2.4/tutorial/generate-key-file/)"
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
+objects:
+  # A headless service to create DNS records
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      # It must match with PetSet.spec.serviceName and MONGODB_SERVICE_NAME
+      name: mongodb
+    spec:
+      # for a headless service proxying is not required
+      clusterIP: None
+      # the list of ports that are exposed by this service
+      ports:
+        - port: 27017
+          name: mongodb
+      # will route traffic to pods having labels matching this selector
+      selector:
+        app: mongo
+  - apiVersion: apps/v1alpha1
+    kind: PetSet
+    metadata:
+      name: replica
+    spec:
+      # pets get DNS/hostnames that follow the pattern: ${metadata.name}-NUM.${spec.serviceName}.default.svc.cluster.local
+      # It must match with Service.metadata.name and MONGODB_SERVICE_NAME
+      serviceName: mongodb
+      replicas: 3
+      # describes the pod that will be created if insufficient replicas are detected
+      template:
+        metadata:
+          # this label will be used for count running pods
+          labels:
+            app: mongo
+          annotations:
+            # 'false' pauses a petset after creation of each pet, to avoid it we set it to 'true'
+            pod.alpha.kubernetes.io/initialized: "true"
+        spec:
+          containers:
+            - name: mongo-container
+              image: "openshift/mongodb-24-centos7"
```

Comment:
>that's a significant limitation we need to make note of.

@bparees I don't fully understand what I should do? Create an issue or document it somehow?